src/Shared/PlatformSupport/AssetLoader.cs

```diff
@@ -67,7 +67,20 @@ public bool Exists(Uri uri, Uri baseUri = null)
         /// <exception cref="FileNotFoundException">
         /// The resource was not found.
         /// </exception>
-        public Stream Open(Uri uri, Uri baseUri = null)
+        public Stream Open(Uri uri, Uri baseUri = null) => OpenWithAssembly(uri, baseUri).Item2;
+        
+        /// <summary>
+        /// Opens the resource with the requested URI.
+        /// </summary>
+        /// <param name="uri">The URI.</param>
+        /// <param name="baseUri">
+        /// A base URI to use if <paramref name="uri"/> is relative.
+        /// </param>
+        /// <returns>An assembly (optional) and a stream containing the resource contents.</returns>
+        /// <exception cref="FileNotFoundException">
+        /// The resource was not found.
+        /// </exception>
+        public Tuple<Assembly, Stream> OpenWithAssembly(Uri uri, Uri baseUri = null)
```

Comment:
Hmm this feels a bit hacky, but I'm not sure what a better way to do it might be. It feels like there should be a parsing step before calling `Open`, but this would be another PR...

Maybe it's just the name? `OpenWithAssembly` sounds like you're _passing_ an assembly to use. Maybe `OpenAndGetAssembly`? Not sure.